src/connection/connection_cl.c

```diff
@@ -1027,8 +1029,15 @@ css_connect_to_cubrid_server (char *host_name, char *server_name)
 
   timeout = prm_get_integer_value (PRM_ID_TCP_CONNECTION_TIMEOUT) * 1000;
 
+  /*
+   * The packing of the server name and type is done by setting the first
+   * character to the numer related to the server type enum value, the rest of the
+   * buffer space is used to copy the server name.
+   */
+  msg = ((char) server_type) + '0';
+  msg.append (server_name, std::strlen (server_name));
```

Comment:
css_server_connect calls `strlen` again on `msg.c_str()`, therefore msg must end with a null-terminated character. Appending `strlen` will miss the ending null-terminated character. Either append `strlen+1` or append the `server_name` without any length mentioned to automatically copy until the null-terminated character.